src/payload.rs

```diff
@@ -1561,5 +1799,77 @@ mod test {
             HashSet::from_iter(vec![1, 3].into_iter()),
         );
     }
-}
 
+    #[test]
+    /// Tests the order of data in a snapshot.
+    ///
+    /// Creates a snapshot via a builder and then checks that the various
+    /// iterators produce correct output.
+    ///
+    /// This assumes that `RouteOrigin` and `RouterKey` order correctly.
+    fn snapshot_order() {
```

Comment:
Perhaps pedantic and highly unlikely, but it seems that if for some reason the `iter`s and `filter_map .. collect`s result in empty collections, the tests will pass. Simple counters in the for loops could be used to test we have seen indeed 4 origins and 3 router keys.

Reply:
There are asserts checking that `orgins_iter` and `keys_iter` are empty after we went over the collected vecs. Doesn’t that – together with unwrapping the next call during the for loop – ensure that the collected vecs have the exact same size as the reference vecs?